### PR TITLE
Scan frequency Implementation

### PR DESCRIPTION
**Pull request by ruflin**
See https://github.com/elastic/filebeat/issues/19

### CODE REVIEW

**Comment by ruflin** [comment]
This change also closes https://github.com/elastic/filebeat/issues/14

File: crawler/prospector_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+	prospector := Prospector{
+		FileConfig: fileConfig,
+	}
+
+	assert.NotNil(t, prospector)
+
+	prospector.Init()
+
+	// Default values expected
```
Comment (tsg) [inline crawler/prospector_test.go]: Is this comment right? Seems like the code checks non-default values.

File: tests/files/config.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ filebeat:
         review: 1
         type: log
       ignoreolder: 24h
+      scanfrequency: 10s
```
Comment (tsg) [inline tests/files/config.yml]: I see most config file options use camelCase, but `ignoreolder` and `scanfrequency` do not. Any reason for this?

File: crawler/prospector.go [hunk 3]
```diff
@@ -44,8 +46,27 @@ func (p *Prospector) Init() {
 		p.FileConfig.IgnoreOlderDuration = cfg.DefaultIgnoreOlderDuration
 	}
 
+	if p.FileConfig.ScanFrequency != "" {
+
+		var err error
+		// Default ignore time time
```
Comment (tsg) [inline crawler/prospector.go]: This comment is unclear to me.

Comment (ruflin) [inline crawler/prospector.go]: I removed the comment completely as I think the code is self explanatory. Same comment was also above, so it seems to come from copy-paste.

File: crawler/prospector_test.go (hunk 1 above)
Comment (ruflin) [inline crawler/prospector_test.go]: Changed the comment. You are right of course.

File: tests/files/config.yml (hunk 2 above)
Comment (ruflin) [inline tests/files/config.yml]: The reason is that this one is tested and the others are not :-) I would prefer to have scanFrequency, but when it is `scanFrequency` in the config file, it is not correctly mapped to go objects. Do you know any trick here?

Comment (tsg) [inline tests/files/config.yml]: yeah, you can use struct tags, like this:
<pre>
type FileConfig struct {
	Paths                 []string
	Fields                map[string]string
	Input                 string
	IgnoreOlder           string `yaml:"ignoreOlder"`
	IgnoreOlderDuration   time.Duration
	ScanFrequency         string `yaml:"scanFrequency"`
	ScanFrequencyDuration time.Duration
}
</pre>

Comment (ruflin) [inline tests/files/config.yml]: I opened an issue for it to make sure we don't forget about it: https://github.com/elastic/filebeat/issues/23

**Pull request merged**